transformations/ocr_perturbation/transformation.py

```diff
@@ -0,0 +1,177 @@
+        :param custom_params: image rendering parameters
+        """
+        assert language in self.languages
+        
+        super().__init__(seed, max_outputs=max_outputs)
+        self.language = language
+        self.params = params
+        self.nlp = self._get_spacy_model()
+
+
+    def generate(self, sentence: str) -> List[str]:
+
+        random.seed(self.seed)
+
+        perturbed_sentences = []        
+        
+        with PyTessBaseAPI(lang=tess_lang[self.language], psm=PSM.RAW_LINE, 
+            oem=OEM.LSTM_ONLY) as ocr:
+            
+            for idx in range(self.max_outputs):
+            
+                perturbed_sentence = ""
+                
+                doc = self.nlp(sentence)
+                assert doc.has_annotation("SENT_START")
+                
+                for sent in doc.sents:
+
+                    image_generator = self._get_image_generator(sent.text)                        
+                    img, lbl = image_generator.next()
+                
+                    ocr.SetImage(img)
+
+                    try:
+                        recognized_text = ocr.GetUTF8Text()        
+                        recognized_text = recognized_text.strip().replace("\n", "").replace("\t", "")
+                    except RuntimeError:
+                        recognized_text = sent.text
+                    
+                    perturbed_sentence += " " + recognized_text
+
+                perturbed_sentences.append(perturbed_sentence.strip())
+                    
+        return perturbed_sentences                
+        
+    def _get_image_generator(self, sentence):
+        return GeneratorFromStrings(
+                [sentence],
+                language=trdg_lang[self.language],
+                size=32,
+                skewing_angle=0,
+                random_skew=False,
+                blur=self.params.blur, 
+                random_blur=False,
+                is_handwritten=False,
+                background_type=self.params.background_type, 
+                distorsion_type=self.params.distorsion_type, 
+                alignment=1, 
+                text_color="#282828",
+                distorsion_orientation=self.params.distortion_orientation,
+                orientation=0, 
+                space_width=1,
+                character_spacing=0,
+                margins=(2,2,2,2)
+            )
+            
+    def _get_spacy_model(self):
+        if self.language == 'en':
+            return spacy_nlp if spacy_nlp else spacy.load("en_core_web_sm")
+        elif self.language == 'fr':
+            return spacy.load("fr_core_news_sm")
+        elif self.language == 'es':
+            return spacy.load("es_core_news_sm")
+        elif self.language == 'de':
+            return spacy.load("de_core_news_sm")
+        else:
+            return None
+
+"""
+# Sample code to demonstrate usage. Can also assist in adding test cases.
```

Comment:
I would add the Sample code to a separate file and add a reference to it in the `README.md`.

Reply:
Done. Thank you @Saad-Mahamood for your feedback!